src/saltext/vmware/modules/esxi.py

```diff
@@ -1166,6 +1166,246 @@ def get_user(
+
+    force
+        Forcefully remove a role even when in use. Default False. (optional).
+
+    esxi_host_name
+        ESXi host name to use for creating the connection. (optional).
+
+    esxi_user_name
+         ESXi username to use for creating the connection. (optional).
+
+    esxi_user_password
+         Password to use for creating the connection. (optional).
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.remove_role role_name=foo
+    """
+    log.debug("Running vmware_esxi.update_role")
+    if not service_instance:
+        service_instance = get_service_instance(
+            opts=__opts__,
+            pillar=__pillar__,
+            esxi_user=esxi_user_name,
+            esxi_host=esxi_host_name,
+            esxi_password=esxi_user_password,
+        )
+    try:
+        role = get_role(role_name=role_name, service_instance=service_instance)
+        if not role:
+            raise salt.exceptions.SaltException("Role {} not found".format(role_name))
+        service_instance.content.authorizationManager.RemoveAuthorizationRole(
+            roleId=role["role_id"], failIfUsed=force
+        )
+        return True
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.AlreadyExists,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def get_role(
+    role_name,
+    esxi_host_name=None,
+    esxi_user_name=None,
+    esxi_user_password=None,
+    service_instance=None,
+):
+    """
+    Get local role on ESXi host.
+
+    role_name
+        Retrieve this role on ESXi host. (required).
+
+    esxi_host_name
+        ESXi host name to use for creating the connection. (optional).
+
+    esxi_user_name
+         ESXi username to use for creating the connection. (optional).
+
+    esxi_user_password
+         Password to use for creating the connection. (optional).
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.get_role role_name=foo
+    """
+    log.debug("Running vmware_esxi.get_role")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(
+            opts=__opts__,
+            pillar=__pillar__,
+            esxi_user=esxi_user_name,
+            esxi_host=esxi_host_name,
+            esxi_password=esxi_user_password,
+        )
+    try:
+        for role in service_instance.content.authorizationManager.roleList:
+            if role.name == role_name:
+                ret["role_id"] = role.roleId
+                ret["role_name"] = role.name
+                ret["privilege_ids"] = list(role.privilege)
+        return ret
```

Comment:
If there is no such role, should we raise an exception?

Reply:
don't want to do that. I am using this for checking with a simple `if` condition for state mods and during update. Otherwise need to catch a salt exception to do the same.